Fix Payer Arg for initial Candy Machine account creation.

tonyboylehub opened the pull request:
User reported that when setting Payer in candymachines `Create` function that it was not using the correct payer.

Turns out payer from the `create()` function wasn't being passed into the pre-account creation for the candy machine which means the default umi identity was trying to fund the account creation which was non deseried behaviour.

**Fix**

Add the payer arg to `createAccount` function if payer is present, if not set undefined which will default back to the umi assigned signer.

```ts
  createAccount(context, {
        payer: newInput.payer  // newly added.  newInput.payer is either Signer | Undefined
        newAccount: newInput.candyMachine,
        lamports,
        space,
        programId: context.programs.get('mplCoreCandyMachineCore').publicKey,
      })
```

tonyboylehub committed:
fixes candymachine creation with correct payer.

tonyboylehub committed:
removed uneeded undefined as already undefined if not present

MarkSackerberg commented on clients/js/src/createCandyMachine.ts:
```diff
@@ -54,6 +54,7 @@ export const createCandyMachine = async (
   return transactionBuilder()
     .add(
       createAccount(context, {
+        payer: newInput.payer,
```
doesn't this need to be `payer: input.payer ?? context.payer,` instead? 
Since we want the umi identity if nothing is provided.

For reference: https://github.com/metaplex-foundation/mpl-core-candy-machine/blob/fba81842958c5f23257b4f4ead1deff4acee45b9/clients/js/src/mintV1.ts#L53

blockiosaurus replied:
If nothing is provided it should be 'undefined' and therefore will just default to the identity already. I don't mind being more explicit though since it's confusing.

tonyboylehub replied:
Yeah if `payer` is undefined the function defaults to assigning the current umi.identity as the payer signer.

I had it as explicit the firsrt time, then I changed it. I'm good either way. I can swap it back?

blockiosaurus replied:
Yes let's be explicit

tonyboylehub committed:
make explicit

tonyboylehub committed:
fix: add idl-build feature to candy-machine-core and candy-guard

Required by anchor 0.30.x for IDL generation via anchor build --idl.

tonyboylehub committed:
fix: exclude idl-build feature from clippy --all-features check

idl-build is an anchor build-system feature not compatible with regular
cargo compilation contexts.

tonyboylehub committed:
ci: install nightly cargo before IDL generation

anchor-cli 0.30.1 uses the nightly toolchain (installed by Solana
platform-tools) internally, but the cargo component is missing from it.

tonyboylehub committed:
fix CI IDL generation: use RUSTUP_TOOLCHAIN instead of nightly install

anchor 0.30.1 respects RUSTUP_TOOLCHAIN to override the nightly toolchain
injected by Solana platform-tools, avoiding proc-macro2 failures on nightly.
